rust/operator-binary/src/env_vars.rs

```diff
@@ -407,23 +419,33 @@ pub fn build_airflow_template_envs(
 
 /// Return environment variables to be applied to the configuration map used in conjunction with
 /// the `kubernetesExecutor` worker: applied to the gitsync `initContainer`.
-pub fn build_gitsync_template(credentials_secret: &Option<String>) -> Vec<EnvVar> {
-    let mut env = vec![];
-
-    if let Some(credentials_secret) = &credentials_secret {
-        env.push(env_var_from_secret(
-            GITSYNC_USERNAME,
-            credentials_secret,
-            "user",
-        ));
-        env.push(env_var_from_secret(
-            GITSYNC_PASSWORD,
-            credentials_secret,
-            "password",
-        ));
+pub fn build_gitsync_template(env_overrides: &HashMap<String, String>) -> Vec<EnvVar> {
+    let mut env: BTreeMap<String, EnvVar> = BTreeMap::new();
```

Comment:
Same as above. Looks like a `Vec` would be enough. Also this block is duplicated. I can live with it,  pointing it out as a dutiful reviewer. :)

Reply:
Thanks! Changed here: [6daed23](https://github.com/stackabletech/airflow-operator/pull/456/commits/6daed23b62813f91fd7d86a2941e83aa37f20e18)